src/ray/gcs/redis_async_context.h

```diff
@@ -33,11 +33,14 @@ typedef void redisCallbackFn(struct redisAsyncContext *, void *, void *);
 namespace ray {
 namespace gcs {
 
+class MockRedisContext;
+
 struct RedisContextDeleter {
   RedisContextDeleter(){};
 
   void operator()(redisContext *context) { redisFree(context); }
   void operator()(redisAsyncContext *context) { redisAsyncFree(context); }
+  void operator()(MockRedisContext *context){};
```

Comment:
Is there a way not to pollute the code with testing code?

Reply:
In this case it seems hard to me... any ideas?